'Select all user's comments' in admin #2027

aguilarm opened the pull request:
## What does this PR do?

Adds #2027 - gmail-like behavior for selecting _all_ of a user's comments in the user detail view of admin. 

## How do I test this PR?

- In admin, select a user
- Click 'Select All'
- A new button appears, 'Select all X comments for USER'
![image](https://user-images.githubusercontent.com/5203478/47578125-319de100-d90e-11e8-815b-f6d9e2e67db7.png)
- Click new button, button switches to:
![image](https://user-images.githubusercontent.com/5203478/47578151-3f536680-d90e-11e8-9da0-5768def79196.png)
- Clicking accept/reject applies new status to _all_ of user's comments

I have one problem, hopefully with some input I can solve it:
- ~~I created a new set of comments to select in the gql query for the UserDetail container that only grabs comment id's and statuses. This then loads all of a user's comments to apply an update. I'm not sure however how to reset it to the first page after running the fetchMore a few times so right now this basically only works once. Perhaps my whole approach here is off.~~ Turns out it I missed default arguments for comment status.

And a couple TODOS:
- ~~I should probably make a fragment to share between the original query and load more query~~
- ~~Once 'Select all by user' is selected, Load More should be disabled until the selection is cleared. Further, once an action is clicked (approve/reject) when 'all by user' is selected, the list should be locked or replaced with a loading indicator, ideally with progress and a cancel button because this will take at least a few seconds.~~ Switched so the element switches to loading state while the batch is running.

Commit:
feat(select-all-for-user): Add behavior to select all comments for a user in admin

Commit:
Merge branch 'master' into admin-bulk-approve-reject

kgardnr commented:
@IAmSamHankins here's a gif of this functionality for your approval!

![bulk_gmail_esque](https://user-images.githubusercontent.com/1077300/47744633-ac9e2900-dc79-11e8-99d6-2fd38dd9034a.gif)

@cvle / @wyattjoh take a look and let us know if it's good code-wise. 

Thanks!

Commit:
fix(batchComments): Set statuses to null explicitly in intial and loadmore calls to fetch ALL comments, add a loading state

aguilarm commented:
I'm pretty sure I found why this was sometimes bugging out and only working once.

It's the CommentsQuery statuses default value being [NONE, ACCEPTED], so if you rejected comments - since my query didn't set a value for statuses - it was not selecting rejected comments to later approve.

Pushed a fix to set statuses to null explicitly and everything looks all set now.

I have it doing 5 comments at a time just for ease of testing with a smaller dataset, but may want to switch that to something like 50 or 100.

Commit:
Merge branch 'master' into admin-bulk-approve-reject

wyattjoh committed:
Merge branch 'master' into admin-bulk-approve-reject

Commit:
refactor(userdetail): Only load one initial comment, switch to 50 comments/batch, create shared fragment

aguilarm commented:
I switched to a shared fragment, load one comment to start and adjusted the # per page to 50.

Still could probably be explicitly setting the status to either [NONE,ACCEPTED] (default), or [Rejected] so I'm not needlessly iterating the entire set every time but usually that's only iterating a handful of extra comments tops - moderation has generally been all-or-nothing when possible so far.

kgardnr committed:
Merge branch 'master' into admin-bulk-approve-reject
